jupyter_book/toc.py

```diff
@@ -114,11 +115,11 @@ def gen_toctree(options, subsections):
 
         # If not a special case, assume we have a "regular" page structure
         if ipage.get("file"):
-            path_sec = ipage.get("file")
+            path_sec = Path(ipage.get("file"))
 
             # Update path so it is relative to the root of the parent
             path_parent_folder = Path(parent_page["file"]).parent
-            path_sec = os.path.relpath(path_sec, path_parent_folder)
+            path_sec = path_sec.relative_to(path_parent_folder).as_posix()
```

Comment:
I think it is safer to use  (relative_to works only for subdirs)

```suggestion
            path_sec = Path(os.path.relpath(path_sec, path_parent_folder)).as_posix()
```